geomesa-spark-sql/geomesa-spark-sql-accumulo/src/main/scala/org/locationtech/geomesa/spark/accumulo/AccumuloSpatialRDDProvider.scala

```diff
@@ -0,0 +1,153 @@
+    try {
+      // get the query plan to set up the iterators, ranges, etc
+      lazy val sft = ds.getSchema(query.getTypeName)
+      lazy val qp = AccumuloJobUtils.getSingleQueryPlan(ds, query)
+
+      if (ds == null || sft == null || qp.isInstanceOf[EmptyPlan]) {
+        sc.emptyRDD[SimpleFeature]
+      } else {
+        val instance = ds.connector.getInstance().getInstanceName
+        val zookeepers = ds.connector.getInstance().getZooKeepers
+
+        val transform = query.getHints.getTransformSchema
+
+        ConfiguratorBase.setConnectorInfo(classOf[AccumuloInputFormat], conf, username, password)
+        if (useMock){
+          ConfiguratorBase.setMockInstance(classOf[AccumuloInputFormat], conf, instance)
+        } else {
+          ConfiguratorBase.setZooKeeperInstance(classOf[AccumuloInputFormat], conf, instance, zookeepers)
+        }
+        InputConfigurator.setInputTableName(classOf[AccumuloInputFormat], conf, qp.table)
+        InputConfigurator.setRanges(classOf[AccumuloInputFormat], conf, qp.ranges)
+        qp.iterators.foreach(InputConfigurator.addIterator(classOf[AccumuloInputFormat], conf, _))
+
+        if (qp.columnFamilies.nonEmpty) {
+          val cf = qp.columnFamilies.map(cf => new org.apache.accumulo.core.util.Pair[Text, Text](cf, null))
+          InputConfigurator.fetchColumns(classOf[AccumuloInputFormat], conf, cf)
+        }
+
+        if (numberOfSplits.isDefined) {
+          GeoMesaConfigurator.setDesiredSplits(conf, numberOfSplits.get * sc.getExecutorStorageStatus.length)
+          InputConfigurator.setAutoAdjustRanges(classOf[AccumuloInputFormat], conf, false)
+          InputConfigurator.setAutoAdjustRanges(classOf[GeoMesaAccumuloInputFormat], conf, false)
+        }
+
+        org.apache.accumulo.core.client.mapreduce.lib.impl.InputConfigurator.setBatchScan(classOf[AccumuloInputFormat], conf, true)
+        org.apache.accumulo.core.client.mapreduce.lib.impl.InputConfigurator.setBatchScan(classOf[GeoMesaAccumuloInputFormat], conf, true)
+        GeoMesaConfigurator.setSerialization(conf)
+        GeoMesaConfigurator.setTable(conf, qp.table)
+        GeoMesaConfigurator.setDataStoreInParams(conf, dsParams)
+        GeoMesaConfigurator.setFeatureType(conf, sft.getTypeName)
+
+        // set the secondary filter if it exists and is  not Filter.INCLUDE
+        qp.filter.secondary
+          .collect { case f if f != Filter.INCLUDE => f }
+          .foreach { f => GeoMesaConfigurator.setFilter(conf, ECQL.toCQL(f)) }
+
+        transform.foreach(GeoMesaConfigurator.setTransformSchema(conf, _))
+
+        // Configure Auths from DS
+        val auths = Option(AccumuloDataStoreParams.authsParam.lookUp(dsParams).asInstanceOf[String])
+        auths.foreach { a =>
+          val authorizations = new Authorizations(a.split(","): _*)
+          InputConfigurator.setScanAuthorizations(classOf[AccumuloInputFormat], conf, authorizations)
+        }
+
```

Comment:
it would be nice if we could replace all this configuration with just the underlying `GeoMesaAccumuloInputFormat.configure` call... we have a ticket for that already